cmd/kops/toolbox_bundle.go

```diff
@@ -0,0 +1,215 @@
+	// Target is the machine we are enrolling in the cluster
+	Target string
+}
+
+func (o *ToolboxBundleOptions) InitDefaults() {
+}
+
+func NewCmdToolboxBundle(f *util.Factory, out io.Writer) *cobra.Command {
+	options := &ToolboxBundleOptions{}
+	options.InitDefaults()
+
+	cmd := &cobra.Command{
+		Use:     "bundle",
+		Short:   toolbox_bundle_short,
+		Long:    toolbox_bundle_long,
+		Example: toolbox_bundle_example,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := RunToolboxBundle(f, out, options, args)
+			if err != nil {
+				exitWithError(err)
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&options.Target, "target", options.Target, "machine to target (IP address)")
+
+	return cmd
+}
+
+func RunToolboxBundle(context Factory, out io.Writer, options *ToolboxBundleOptions, args []string) error {
+	if len(args) == 0 {
+		return fmt.Errorf("Specify name of instance group for node")
+	}
+	if len(args) != 1 {
+		return fmt.Errorf("Can only specify one instance group")
+	}
+
+	if options.Target == "" {
+		return fmt.Errorf("target is required")
+	}
+	groupName := args[0]
+
+	cluster, err := rootCommand.Cluster()
+	if err != nil {
+		return err
+	}
+
+	clientset, err := context.Clientset()
+	if err != nil {
+		return err
+	}
+
+	ig, err := clientset.InstanceGroupsFor(cluster).Get(groupName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error reading InstanceGroup %q: %v", groupName, err)
+	}
+	if ig == nil {
+		return fmt.Errorf("InstanceGroup %q not found", groupName)
+	}
+
+	builder := bundle.Builder{
+		Clientset: clientset,
+	}
+	bundleData, err := builder.Build(cluster, ig)
+	if err != nil {
+		return fmt.Errorf("error building bundle: %v", err)
+	}
+
+	sshUser := os.Getenv("USER")
+
+	nodeSSH := &kutil.NodeSSH{
+		Hostname: options.Target,
+	}
+	nodeSSH.SSHConfig.HostKeyCallback = ssh.InsecureIgnoreHostKey()
+	nodeSSH.SSHConfig.User = sshUser
+	sshIdentity := filepath.Join(os.Getenv("HOME"), ".ssh", "id_rsa")
+	if err := kutil.AddSSHIdentity(&nodeSSH.SSHConfig, sshIdentity); err != nil {
+		return err
+	}
+
+	sshClient, err := nodeSSH.GetSSHClient()
+	if err != nil {
+		return fmt.Errorf("error getting SSH client: %v", err)
+	}
+
+	if err := runSshCommand(sshClient, "sudo mkdir -p /etc/kubernetes/bootstrap"); err != nil {
+		return err
+	}
+
+	root, err := nodeSSH.Root()
+	if err != nil {
+		return fmt.Errorf("error connecting to nodeSSH: %v", err)
+	}
+	for _, file := range bundleData.Files {
+		sshAcl := &vfs.SSHAcl{
+			Mode: file.Header.FileInfo().Mode(),
+		}
+		p := root.Join("etc", "kubernetes", "bootstrap", file.Header.Name)
+		glog.Infof("writing %s", p)
+		if err := p.WriteFile(file.Data, sshAcl); err != nil {
+			return fmt.Errorf("error writing file %q: %v", file.Header.Name, err)
+		}
+	}
+
+	if err := runSshCommand(sshClient, "sudo /etc/kubernetes/bootstrap/bootstrap.sh"); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func runSshCommand(sshClient *ssh.Client, cmd string) error {
```

Comment:
Nit pick, this should go into a ssh package, or bare metal package. We can refactor.

Reply:
Yes, agreed... a cleanup of SSH is definitely needed.  Trying to keep the PRs relatively isolated though - although this one crept into the apply logic.